src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -544,19 +572,26 @@ export class HttpConnection implements IConnection {
     }
 
     private _resolveNegotiateUrl(url: string): string {
-        const index = url.indexOf("?");
-        let negotiateUrl = url.substring(0, index === -1 ? url.length : index);
-        if (negotiateUrl[negotiateUrl.length - 1] !== "/") {
-            negotiateUrl += "/";
+        const negotiateUrl = new URL(url);
+
+        if (negotiateUrl.pathname.endsWith('/')) {
+            negotiateUrl.pathname += "negotiate";
+        } else {
+            negotiateUrl.pathname += "/negotiate";
         }
-        negotiateUrl += "negotiate";
-        negotiateUrl += index === -1 ? "" : url.substring(index);
+        const searchParams = new URLSearchParams(negotiateUrl.searchParams);
 
-        if (negotiateUrl.indexOf("negotiateVersion") === -1) {
-            negotiateUrl += index === -1 ? "?" : "&";
-            negotiateUrl += "negotiateVersion=" + this._negotiateVersion;
+        if (!searchParams.has("negotiateVersion")) {
+            searchParams.append("negotiateVersion", this._negotiateVersion.toString());
         }
-        return negotiateUrl;
+
+        if (this._options._useStatefulReconnect === true && !searchParams.has("useAck")) {
```

Comment:
What happens if the original negotiateUrl has `useAck` but `this._options._useStatefulReconnect` is not configured? Will the server and client end up in different modes?

Reply:
Oh, good catch. We should either remove it or set the option to true for the user.